java/simulated_ble/README.md

```diff
@@ -0,0 +1,306 @@
+| ------------------------ | -------------------------------------- | ---------------------- | ---------------------- |
+| Temperature data message | Parse and construct a new JSON message | Structure JSON message | `ConverterModule.java` |
+
+This module is a typical Azure IoT Edge module. It accepts temperature message from another module (a hardware module, or in this case our simulated BLE module); and then it normalizes the temperature message to a structured JSON message (including appending the message ID, setting the property of whether we need to trigger the temperature alert, and so on).
+
+```java
+@Override
+public void receive(Message message) {
+  try {
+    float temperature = Float.parseFloat(new String(message.getContent()));
+    Map<String, String> inputProperties = message.getProperties();
+
+    HashMap<String, String> properties = new HashMap<>();
+    properties.put("source", inputProperties.get("source"));
+    properties.put("macAddress", inputProperties.get("macAddress"));
+    properties.put("temperatureAlert", temperature > 30 ? "true" : "false");
+
+    String content = String.format(
+      "{ \"deviceId\": \"Intel NUC Gateway\", \"messageId\": %d, \"temperature\": %f }",
+      ++this.messageCount, temperature);
+
+    this.publish(new Message(content.getBytes(), properties));
+  } catch (Exception ex) {
+    ex.printStackTrace();
+  }
+}
+```
+
+### Printer Module
+
+| Input                          | Processor | Output                     | Source File          |
+| ------------------------------ | --------- | -------------------------- | -------------------- |
+| Any message from other modules | N/A       | Log the message to console | `PrinterModule.java` |
+
+This module is the simplest one among all the modules we create. Let's just show the code without explaining it.
+
+```java
+@Override
+public void receive(Message message) {
+  System.out.println(message.toString());
+}
+```
+
+### Azure IoT Edge Configuration
+
+The final step before running the modules is to configure the Azure IoT Edge and to establish the connections between modules.
+
+First we need to declare our Java loader (since Azure IoT Edge supports loaders of different languages) which could be referenced by its `name` in the sections afterward.
+
+```json
+"loaders": [{
+  "type": "java",
+  "name": "java",
+  "configuration": {
+    "jvm.options": {
+      "library.path": "./"
+    }
+  }
+}]
+```
+
+We are also required to declare modules. Similar to the loaders, they can also be referenced by the `name`. In the declaration of each module, we need to specify the loader (which should be the one we defined before) as well as the entry-point (should be the normalized class name of our module). Let's take the `SimulatedBleModule` as an example (sometimes even `args` is `null`, but you have to put it in the JSON file, otherwise it will cause a failure):
```

Comment:
Once we have declared our loaders we will also need to declare our modules as well. Similar to declaring the loaders, they can also be referenced by their `name` attribute. When declaring a module, we need to specify the loader it should use (which should be the one we defined before) and the entry-point (should be the normalized class name of our module) for each module. Let's take the `SimulatedBleModule` as an example (sometimes even `args` is `null`, but you have to put it in the JSON file, otherwise it will cause a failure):